pkg/provisioner/ironic/ironic.go

```diff
@@ -1348,3 +1369,74 @@ func (p *ironicProvisioner) PowerOff() (result provisioner.Result, err error) {
 
 	return result, nil
 }
+
+// Retrieve RAID configuration in form of Ironic's clean steps
+func (p *ironicProvisioner) buildCleanSteps() (cleanSteps []nodes.CleanStep) {
+	raidCleanSteps := p.buildRAIDCleanSteps(&(p.host.Spec.RAID))
+	return append(cleanSteps, raidCleanSteps...)
+}
+
+// buildRAIDCleanSteps build the Ironic's clean steps with arguments the given RAID configuration
+func (p *ironicProvisioner) buildRAIDCleanSteps(raidConfig *metal3v1alpha1.RAIDConfig) (cleanSteps []nodes.CleanStep){
+	if &raidConfig == nil {
+		return
+	}
+	logicalDisks := []map[string]interface{}{}
+	if raidConfig.RootVolume != nil {
+		rootVolumeArgs := p.buildRAIDVolumeArgs(raidConfig.RootVolume, true)
+		logicalDisks = append(logicalDisks, rootVolumeArgs)
+	}
+	for _, volume := range raidConfig.Volumes {
+		volumeArgs := p.buildRAIDVolumeArgs(&volume, false)
+		logicalDisks = append(logicalDisks, volumeArgs)
+	}
+	if len(logicalDisks) > 0 {
```

Comment:
Do we have a way to request removal of RAID configuration?

Reply:
Currently, there is no option to request RAID removal in `provisioning` phase. I plan to put RAID configuration removal in `deprovisioning` state

Reply:
We're going to need to ensure we delete all existing RAID config before applying the new settings. On line 1401 this passes a "delete_configuration" instruction. Does that do what I'm describing? If so, we need to *always* pass it, regardless of whether there is new RAID configuration to give. We also need to pass it during deprovisioning.